### PR TITLE
Revert "Revert "Fix permissions on cluster deletion in az cli""

### PR DESCRIPTION
**Pull request by ehashman**
This reverts commit 45baf2f5b92f1b7e17919856131fe4bc7838e260.

### Which issue this PR addresses:

Reopening to continue on discussion in #1011.

### CODE REVIEW

File: python/az/aro/azext_aro/custom.py [hunk 1]
```diff
@@ -128,9 +135,56 @@ def aro_create(cmd,  # pylint: disable=too-many-locals
...
+            logger.warning(
+                'Unable to retrieve the cluster service principal. This means '
+                'that it may have been deleted, and some cleanup may fail.'
+            )
+
+        rp_client_id = os.environ.get('AZURE_FP_CLIENT_ID', FP_CLIENT_ID)
+
+        rp_client_sp = aad.get_service_principal(rp_client_id)
+        if rp_client_sp:
+            sp_ids.append(rp_client_sp.object_id)
```
Comment (ehashman) [inline python/az/aro/azext_aro/custom.py:178]: So @jim-minter I went to implement your suggested changes, but as far as I can tell, the code as written here does not introduce a regression that would prevent deletion if there's some issue with the RP SPN? In both the client and RP SPN cases, if we can't get them, we don't append them to the list to tweak perms for.

Is the issue that we may be able to retrieve the SPN but its permissions would be reduced?

Do we just want to `try:` the block below and pass if this fails, similar to the case wherein we can't get the cluster to try this?

Comment (jim-minter) [reply python/az/aro/azext_aro/custom.py:178]: @ehashman if the caller doesn't have permissions to call get_service_principal (example: the caller is a service principal with minimal AAD permissions), get_service_principal will raise an exception and the delete will not be called.  In our environment, $AZURE_RP_CLIENT_ID is such an SP.

**Comment by ehashman** [comment]
Just need to do one manual integration test on this once it passes CI.

**Comment by ehashman** [comment]
I need to fix the Python flaking errors, but I am also seemingly not able to test this as the `az aro` dev extension is getting suppressed, even though my local version should be higher than the suppression version. @dkorzuno are there any doc changes/setup I need to do now that this change has merged?

**Comment by dkorzuno** [comment]
> I need to fix the Python flaking errors, but I am also seemingly not able to test this as the `az aro` dev extension is getting suppressed, even though my local version should be higher than the suppression version. @dkorzuno are there any doc changes/setup I need to do now that this change has merged?

Hmm... I don't recall anything in particular. Once I bumped the extension version to 1.0.1 it worked for me. Conversely, as soon as I made the version 0.1.0 it got suppressed.

**Comment by mjudeikis** [comment]
@dkorzuno did we upstreamed this? Or this is still in our repo only? We need to make sure this is fixes asap

File: python/az/aro/azext_aro/custom.py [hunk 2]
```diff
@@ -128,9 +135,42 @@ def aro_create(cmd,  # pylint: disable=too-many-locals
...
+        if rp_client_sp:
+            sp_id = rp_client_sp.object_id
+            assign_contributor_to_vnet(cmd.cli_ctx, vnet, sp_id)
+            assign_contributor_to_routetable(cmd.cli_ctx,
+                                             worker_subnets | {master_subnet},
+                                             sp_id)
+    except (CloudError, HttpOperationError) as e:
+        # Default to old deletion behaviour in case operations throw an
+        # exception above.
+        pass
```
Comment (jim-minter) [inline python/az/aro/azext_aro/custom.py:172]: I'd like to see the error logged in such a way that there is *some* way to see it if we need to (--debug, whatever) and that it doesn't worry people and cause them to ask for support.

**Review by jim-minter — APPROVED** [review]
One comment, then lgtm (assuming you've been able to test it)

**Comment by ehashman** [comment]
Tested and _should_ be gtg now.

**Pull request merged**